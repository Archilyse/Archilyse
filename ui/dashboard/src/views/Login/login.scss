@import '../../common/theme.scss';

$login-button-width: 323px;
$website-brand-color: #21406c;

.login {
  min-width: 100vw;
  min-height: 100vh;
  background: $website-brand-color;
  background-size: cover;

  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;

  .greeting {
    display: flex;
    flex-direction: column;
    width: 541px;

    h1 {
      color: white;
      font-weight: 500;
      font-size: 3.75rem; // 60px;
      line-height: 72px;
    }
    .subtitle {
      margin-top: 0;
      font-weight: 300;
    }
  }

  form {
    position: relative;
    top: $unit * 4;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    @media #{$extra-large-screen} {
      margin-left: $unit * 20;
    }
  }

  .login-input {
    @include large-input(white);
    box-shadow: 0px 4px 7px rgba(0, 0, 0, 0.16);
    background-color: transparent;
    color: white;

    & input::placeholder {
      color: white !important;
      font-family: 'Barlow';
    }
  }

  a {
    font-size: $smaller-font-size;
    line-height: 14px;
    text-align: right;
    width: $login-button-width;
    margin-top: $unit;
    @include unstyle-link(white);
  }

  button.extra-large {
    width: $login-button-width;
    @media #{$extra-large-screen} {
      width: $login-button-width * 2;
    }
  }
}
