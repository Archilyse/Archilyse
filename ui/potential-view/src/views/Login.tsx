import React from 'react';
import { Form, PublicFormsContainer, useLogin } from 'archilyse-ui-components';
import { C } from 'Common';
import { ActionsType, checkAccess } from '../common/roles';

const FIELDS = [
  { name: 'user', label: 'User', required: true },
  { name: 'password', label: 'Password', type: 'password', required: true },
];

const Login = () => {
  const { handleSubmit } = useLogin({
    getInitialPageUrl: C.URLS.HOME,
    checkAccess: intent => checkAccess(intent as ActionsType),
  });

  return (
    <PublicFormsContainer appTitle="Potential Simulations">
      <h2 className="public-forms-title">Sign in</h2>
      <Form
        fields={FIELDS}
        onSubmit={handleSubmit}
        SubmitButton={() => (
          <button className="secondary-button large" type="submit">
            Sign in
          </button>
        )}
      />
    </PublicFormsContainer>
  );
};

export default Login;
