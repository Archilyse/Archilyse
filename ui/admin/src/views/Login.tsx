import React from 'react';
import { Form, PublicFormsContainer, useLogin } from 'archilyse-ui-components';
import { getInitialPageUrlByRole } from '../common/modules';
import { ActionsType, checkAccess } from '../common/roles';
import fields from '../common/forms/login';
import './login.scss';

const ADMIN_BACKGROUND_COLOR = '#445f86';

const Login = () => {
  const { handleSubmit } = useLogin({
    getInitialPageUrl: getInitialPageUrlByRole,
    checkAccess: intent => checkAccess(intent as ActionsType),
  });

  return (
    <PublicFormsContainer appTitle="Admin" backgroundColor={ADMIN_BACKGROUND_COLOR}>
      <h2 className="public-forms-title">Sign in</h2>
      <Form
        fields={fields}
        onSubmit={handleSubmit}
        SubmitButton={() => (
          <button className="secondary-button large" type="submit">
            Sign in
          </button>
        )}
      />
    </PublicFormsContainer>
  );
};

export default Login;
