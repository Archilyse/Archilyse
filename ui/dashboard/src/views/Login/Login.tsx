import React, { useState } from 'react';
import { useLogin } from 'archilyse-ui-components';
import { C } from '../../common';
import { ActionsType, checkAccess } from '../../common/roles';
import getInitialPageUrlByRole from '../../common/modules/getInitialPageUrlByRole';
import './login.scss';

const Login = () => {
  const [user, setUser] = useState('');
  const [password, setPassword] = useState('');

  const { handleSubmit } = useLogin({
    getInitialPageUrl: getInitialPageUrlByRole,
    checkAccess: intent => checkAccess(intent as ActionsType),
  });

  const onSubmit = (event: React.FormEvent) => {
    event.preventDefault();
    handleSubmit({ user, password });
  };

  const Greeting = () => {
    return (
      <div className="greeting">
        <h1>
          Welcome to <br />
          Archilyse.
        </h1>
        <h1 className="subtitle">Please sign in.</h1>
      </div>
    );
  };

  return (
    <div className="login">
      <Greeting />
      <form onSubmit={onSubmit}>
        <input
          id="user"
          name="user"
          placeholder="Username"
          className="login-input"
          required
          onChange={event => setUser(event.target.value)}
          value={user}
        />
        <input
          id="password"
          name="password"
          placeholder="Password"
          className="login-input"
          type="password"
          required
          onChange={event => setPassword(event.target.value)}
          value={password}
        />
        <button type="submit" className="secondary-button extra-large">
          Sign in
        </button>
        <a href={C.URLS.ARCHILYSE_CONTACT()} target="_blank" rel="noreferrer">
          Get in contact
        </a>
      </form>
    </div>
  );
};
export default Login;
