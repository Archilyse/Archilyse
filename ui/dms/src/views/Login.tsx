import React from 'react';
import { Form, PublicFormsContainer, useLogin } from 'archilyse-ui-components';
import { Link } from 'react-router-dom';
import { C } from '../common';

import { getInitialPageUrlByRole } from '../common/modules';
import { ActionsType, checkAccess } from '../common/roles';
import fields from '../common/forms/login';
import './login.scss';

const Login = () => {
  const { handleSubmit } = useLogin({
    getInitialPageUrl: getInitialPageUrlByRole,
    checkAccess: intent => checkAccess(intent as ActionsType),
  });

  return (
    <PublicFormsContainer>
      <h2 className="public-forms-title">Sign in</h2>
      <p>If you have an account with us, please sign in</p>
      <Form
        id={'login-user-form'}
        fields={fields}
        onSubmit={handleSubmit}
        SubmitButton={() => (
          <button id="login-dms-user" className="secondary-button large" type="submit">
            Sign in
          </button>
        )}
        separatedLabels
      />
      <Link to={C.URLS.FORGOT_PASSWORD()} className="forgot-password-link">
        Forgot password?
      </Link>
    </PublicFormsContainer>
  );
};

export default Login;
