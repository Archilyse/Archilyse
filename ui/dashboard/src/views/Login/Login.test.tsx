import * as React from 'react';
import { cleanup, render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Login from './Login';

afterEach(cleanup);

it('renders correctly', () => {
  const { container } = render(
    <BrowserRouter>
      <Login />
    </BrowserRouter>
  );
  expect(container).toMatchSnapshot();
});
