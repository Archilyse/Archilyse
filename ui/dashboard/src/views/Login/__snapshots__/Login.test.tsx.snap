// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="login"
  >
    <div
      class="greeting"
    >
      <h1>
        Welcome to 
        <br />
        Archilyse.
      </h1>
      <h1
        class="subtitle"
      >
        Please sign in.
      </h1>
    </div>
    <form>
      <input
        class="login-input"
        id="user"
        name="user"
        placeholder="Username"
        required=""
        value=""
      />
      <input
        class="login-input"
        id="password"
        name="password"
        placeholder="Password"
        required=""
        type="password"
        value=""
      />
      <button
        class="secondary-button extra-large"
        type="submit"
      >
        Sign in
      </button>
      <a
        href="https://www.archilyse.com/contact/"
        rel="noreferrer"
        target="_blank"
      >
        Get in contact
      </a>
    </form>
  </div>
</div>
`;
